### PR TITLE
NODE-2951: Don't auto destroy read stream on Node 14+

### PR DESCRIPTION
**Pull request by durran**
The new autoDestroy option for streams in Node 14+ would emit an extra close event when the stream has ended.

### CODE REVIEW

File: test/functional/cursor.test.js [hunk 1]
```diff
@@ -1550,6 +1550,50 @@ describe('Cursor', function () {
     }
   });
 
+  it('does not auto destroy streams', {
+    metadata: {
+      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
```
Comment (nbbeeken) [inline test/functional/cursor.test.js:1555]: I know this `topology` setting is what the other tests have, but for some time the valid topologies have been paired down to the first three in this list: https://github.com/mongodb/node-mongodb-native/blob/a516903bf91e34bb83a3bf565c2a39cdbfddd072/test/tools/runner/filters/mongodb_topology_filter.js#L47-L55

So you can omit this filter all together

File: test/functional/cursor.test.js [hunk 2]
```diff
@@ -1550,6 +1550,50 @@ describe('Cursor', function () {
...
+
+      for (var i = 0; i < 10; i++) {
+        docs.push({ a: i + 1 });
+      }
+
+      const configuration = this.configuration;
+      const client = configuration.newClient(configuration.writeConcernMax(), { maxPoolSize: 1 });
+      client.connect((err, client) => {
+        expect(err).to.not.exist;
+        //this.defer(() => client.close());
```
Comment (nbbeeken) [inline test/functional/cursor.test.js:1569]: cruft? I see the close in the end handler below if that was intended.

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: test/functional/cursor.test.js (hunk 2 above)
Comment (durran) [reply test/functional/cursor.test.js]: Updated.

File: test/functional/cursor.test.js (hunk 1 above)
Comment (durran) [reply test/functional/cursor.test.js]: Oh thanks for that - I've removed now.

**Review by nbbeeken — APPROVED** [review]
lgtm!

**Pull request merged**